messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/JMSQueueRemove.java

```diff
@@ -56,18 +54,16 @@ private JMSQueueRemove() {
     protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
 
         final ServiceName serviceName = MessagingServices.getActiveMQServiceName(PathAddress.pathAddress(operation.get(ModelDescriptionConstants.OP_ADDR)));
+        final ServiceName jmsServiceName = JMSServices.getJmsManagerBaseServiceName(serviceName);
         final PathAddress address = PathAddress.pathAddress(operation.require(OP_ADDR));
         final String name = address.getLastElement().getValue();
 
-        ServiceController<?> service = context.getServiceRegistry(false).getService(serviceName);
-        ActiveMQServer server = ActiveMQServer.class.cast(service.getValue());
-        JMSServerControl control = JMSServerControl.class.cast(server.getManagementService().getResource(ResourceNames.JMS_SERVER));
-        if (control != null) {
-            try {
-                control.destroyQueue(name, true);
-            } catch (Exception e) {
-                throw new OperationFailedException(e);
-            }
+        ServiceController<?> service = context.getServiceRegistry(false).getService(jmsServiceName);
```

Comment:
This should be getServiceRegistry(true).

It doesn't have a practical effect because this op would have already gotten the exclusive op lock in Stage.RUNTIME, but it's good as a form of illustration to have our code always use 'true' if the code is accessing MSC with the intent to modify runtime state.

Reply:
fixed (by amending the WFLY-9407 commit).

I misunderstood the `getServiceRegistry(boolean)` javadoc. it says:
> * @param modify {@code true} if the operation may be modifying a service, {@code false} otherwise
     
I understand this as passing `true` if the service *itself* is modified (e.g. calling `setMode`, `{add|remove}Listener`). Maybe we should rephrase: "if the runtime state of a service is modified".

Reply:
@jmesnil Yeah, that keeps being an issue. https://github.com/wildfly/wildfly-core/pull/2964